engine/class_modules/sc_shaman.cpp

```diff
@@ -139,6 +139,16 @@ static std::vector<player_t*>& __check_distance_targeting( const action_t* actio
 typedef std::pair<std::string, simple_sample_data_with_min_max_t> data_t;
 typedef std::pair<std::string, simple_sample_data_t> simple_data_t;
 
+template <typename Action, typename Actor, typename... Args>
+action_t* get_action( util::string_view name, Actor* actor, Args&&... args )
+{
+  action_t* a = actor->find_action( name );
+  if ( !a )
+    a = new Action( name, actor, std::forward<Args>( args )... );
+  assert( dynamic_cast<Action*>( a ) && a->name_str == name && a->background );
+  return a;
+}
```

Comment:
Doesn't the shaman module have an existing way to get secondary actions?

Reply:
I do see `void trigger_secondary_ability()` but I didn't spot a way to get the action itself. That said, I noticed some other spells that set `tick_action` to an instance of some other spell directly, so I've done so in 397c962ae2 and that appears to work fine, so I've removed the `get_action` stuff borrowed from the mage module.